NTC-Weather-Forecast-COVID-19/NTC-Weather-Forecast-COVID-19/Control/Weather Forecast/WeatherViewController.swift

```diff
@@ -9,26 +9,116 @@ import UIKit
     
-    @IBOutlet private weak var cityLable: UILabel!
-    @IBOutlet private weak var countryLable: UILabel!
-    @IBOutlet private weak var timeLable: UILabel!
+    @IBOutlet private weak var cityLabel: UILabel!
+    @IBOutlet private weak var countryLabel: UILabel!
+    @IBOutlet private weak var timeLabel: UILabel!
     @IBOutlet private weak var iconImageView: UIImageView!
-    @IBOutlet private weak var descriptionLable: UILabel!
-    @IBOutlet private weak var tempLable: UILabel!
-    @IBOutlet private weak var humidityLable: UILabel!
-    @IBOutlet private weak var windSpeedLable: UILabel!
+    @IBOutlet private weak var descriptionLabel: UILabel!
+    @IBOutlet private weak var tempLabel: UILabel!
+    @IBOutlet private weak var humidityLabel: UILabel!
+    @IBOutlet private weak var windSpeedLabel: UILabel!
     @IBOutlet private weak var sevenWeatherTableView: UITableView!
+    private var city = CityWorld()
+    private var sevenDaysWeather = SevenDaysWeather()
+    private var currentWeather = CurrentWeather()
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        setup()
+        loadAPI()
     }
     
-    @IBAction func click_SettingButton(_ sender: UIBarButtonItem) {
+    private func setup() {
+        sevenWeatherTableView.dataSource = self
     }
     
-    @IBAction func click_GoToSearchCityScreenButton(_ sender: UIBarButtonItem) {
+    private func loadAPI() {
+        let lat = Double(city.lat) ?? 0
+        let long = Double(city.long) ?? 0
+        let dispatchGroup = DispatchGroup()
+        dispatchGroup.enter()
+        WeatherAPIService.shared.getCurrentWeatherByLocate(lat: lat, long: long) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let currentWeather):
+                self.currentWeather = currentWeather
+                dispatchGroup.leave()
+            case .failure(let error):
+                print(error)
+            }
+        }
+        
+        dispatchGroup.enter()
+        WeatherAPIService.shared.get7DaysWeather(lat: lat, long: long) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let sevenDaysWeather):
+                self.sevenDaysWeather = sevenDaysWeather
+                dispatchGroup.leave()
+            case .failure(let error):
+                print(error)
+            }
+        }
+        
+        dispatchGroup.notify(queue: .main) { [weak self] in
+            guard let self = self else { return }
+            self.loadData()
+            self.sevenWeatherTableView.reloadData()
+        }
     }
     
-    @IBAction func click_SeeDetailButton(_ sender: UIButton) {
+    private func loadData() {
+        cityLabel.text = city.city
+        countryLabel.text = city.country
+        if currentWeather.weather.isEmpty {
+            var alter = UIAlertController()
```

Comment:
let alert = UIAlertController()

Reply:
vì t có  thay đổi xét alter = alter.showErrorAlter(title: "Notice", message: "Not data weather of this City!") nên cái này sửa thành let nó sẽ báo lỗi.

Reply:
Thế ô đổi lại tên đi sai tên r

Reply:
ok o :)